examples/eventlistener-tls-connection/README.md

```diff
@@ -0,0 +1,102 @@
+## EventListener Secure Connection
+
+Triggers now support both `HTTP` and `HTTPS` connection by adding some configurations to eventlistener.
+
+### Prerequisites
+* Certificates with Key and Cert
+* Secret which includes those certificates
+
+### Try it out locally:
+
+#### Creating Prerequisites
+
+* #### Certificates with Key and Cert.
+
+##### 1. Steps to generate root key, cert
+1. Create Root Key
+   ```text
+   openssl genrsa -des3 -out rootCA.key 4096
+   ```
+2. Create and self sign the Root Certificate
+   ```text
+   openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 1024 -out rootCA.crt
+   ```
+##### 2. Steps to generate certificate (for each server)
+1. Create the certificate key
+   ```text
+   openssl genrsa -out tls.key 2048
+   ```
+2. Create the signing (csr)
+
+* The CSR is where you specify the details for the certificate you want to generate. 
+This request will be processed by the owner of the root key to generate the certificate.
+
+* **Important:** While creating the csr it is important to specify the `Common Name` providing the IP address or domain name for the service, otherwise the certificate cannot be verified.
+   ```text
+   openssl req -new -key tls.key -out tls.csr
+   ```
+3. Generate the certificate using the tls csr and key along with the CA Root key
+   ```text
+   openssl x509 -req -in tls.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out tls.crt -days 500 -sha256
+   ```
+##### 3. Follow same steps from 2 to generate certificates for client also
+
+* #### Create secret which includes those certificates
+   ```text
+   kubectl create secret generic tls-secret-key --from-file=tls.crt --from-file=tls.key
+   ```
+
+#### Configuring EventListener for TLS connection
+```yaml
```

Comment:
Nit: Why not put this in yaml file like other examples? It make it faster to test out examples. Otherwise, lgtm.

Reply:
We can put but the secret will be given by user which contains cert and key files so even added sample example in README so that user can refer

Reply:
We can probably assume `tls-key-secret` as secretKeyRef.

Reply:
if we assume then we should create that before hand and for that we also need to create cert n keys

Reply:
It's OK to not create key beforehand because those command will do it anyway.